app/vmagent/remotewrite/remotewrite.go

```diff
@@ -466,10 +466,19 @@ func tryPush(at *auth.Token, wr *prompbmarshal.WriteRequest, dropSamplesOnFailur
 
 	rowsCount := getRowsCount(tss)
 
-	if *disableOnDiskQueue {
-		// Quick check whether writes to configured remote storage systems are blocked.
-		// This allows saving CPU time spent on relabeling and block compression
-		// if some of remote storage systems cannot keep up with the data ingestion rate.
+	disableOnDiskAll := true
+	for _, rwctx := range rwctxs {
```

Comment:
This check resides in the vmagent "hot path": each call to `tryPush` will have to do the same work again&again.
wdyt about making  `disableOnDiskAll ` a global variable and initiate it only once in, let's say, `Init()` function?

Reply:
That would be much better indeed, let me fix this asap :)